python/metatensor-torch/metatensor/torch/atomistic/model.py

```diff
@@ -0,0 +1,234 @@
+from typing import Dict, List
+
+import torch
+
+from .. import TensorBlock
+from . import ModelCapabilities, ModelRunOptions, NeighborsListOptions, System
+from .units import KNOWN_QUANTITIES, Quantity
+
+
+class MetatensorAtomisticModule(torch.nn.Module):
+    """TODO"""
+
+    # Some annotation to make the TorchScript compiler happy
+    _requested_neighbors_lists: List[NeighborsListOptions]
+    _known_quantities: Dict[str, Quantity]
+
+    def __init__(self, module: torch.nn.Module, capabilities: ModelCapabilities):
```

Comment:
`capabilities` is something like the target? i.e. "forces", "dipole moments", "partial charges"?

Reply:
`capabilities.outputs` contains the different targets the model is able to produce. `capabilities` contains other information about the model (unit used as input, species it can handle).

Maybe we should rename this to `ModelDefinition` or something, and also store in there the model authors, papers to cite, date of training, etc.

Reply:
I also stumbled on this term, I thought about `ModelConfig`, but the term does not capture really that in includes the model outputs.

> Maybe we should rename this to ModelDefinition or something, and also store in there the model authors, papers to cite, date of training, etc.
 
I like `ModelDefinition`, then this would include a metainformation variable of type a `Dict[str, str]` ?

Reply:
I would prefer to give more structure to the data, and have multiple fields of type `str` instead of a dict. We could have a `other` field if people want to store more data in there, but I would start without this for now.

Reply:
Maybe we could consider metadata that is included in models from PyTorch Hub or hugging face
https://pytorch.org/docs/stable/hub.html#torch.hub.load
https://huggingface.co/docs/transformers/v4.34.1/en/model_doc/auto#transformers.AutoModel.from_pretrained

I identify these parameters
- `provider` hub platform (e.g. "Hugging Face")
- `repo_url` (e.g. "https://camembert-model.fr")
- `model_name`  (e.g. "camenbert")
- `model_checkpoint` (e.g. "camembert-large")

Reply:
Might be also useful to consider, the metadata that is contained in ONNX models https://onnx.ai/onnx/intro/concepts.html#metadata I identified these as possibly useful

- __producer_version__: The version of the generating tool.
like version of the model or commit ID
- __model_version__: The version of the model itself, encoded in an integer.
I think this as basically what `model_checkpoint` above is expressing, just that they express it as a string instead of a number. If you look at models from the computer vision community https://pytorch.org/vision/stable/models.html it also makes more sense. For example the first model the `model_name`  would be `AlexNet`  and the version would be `AlexNet_Weights.IMAGENET1K_V1`. I think that would be good enough to track models. Sometimes they use the `_VX` at the end to describe that the training procedure changed a bit, but sometimes they change the name at the beginning to do that. It is a mess.
- __model_license__: The well-known name or URL of the license under which the model is made available.
Not sure if this makes sense, as the license should be stored in the repo where the model comes from I would say
- __doc_string__: Human-readable documentation for this model

Reply:
I'm not sold on the hugging face style metadata, it feels more related to a full repository of models instead of a single one. The ONNX metatadata makes a lot more sense to me.

Reply:
We currently collect some minimal metadata, which should be expanded with stuff like this. If you agree @agoscinski I would leave the metadata definition to a later PR?